python/ray/autoscaler/sdk/sdk.py

```diff
@@ -240,6 +240,24 @@ def request_resources(
         >>> request_resources( # doctest: +SKIP
         ...     bundles=[{"CPU": 1}, {"CPU": 1}, {"CPU": 1}])
     """
+
+    if num_cpus is not None and not isinstance(num_cpus, int):
+        raise TypeError("num_cpu must be either None or an int.")
+    if bundles is not None:
+        if isinstance(bundles, List):
+            for bundle in bundles:
+                if isinstance(bundle, Dict):
+                    if "CPU" not in bundle or "GPU" not in bundle:
+                        raise TypeError("CPU or GPU key should be present in bundle.")
```

Comment:
Actually, any string key is valid. So we should just validate that the keys are strings.

A node can be labelled with custom resource capacities. like "CUSTOM": 10.
Then you can for example
schedule 10 tasks requesting {"CUSTOM": 1} on such a node, or
trigger upscaling of 2 such nodes by requesting two {"CUSTOM": 10} bundles
`ray.autoscaler.sdk.request_resources(bundles=[{"CUSTOM":10}]*2)`